rasa/nlu/featurizers/dense_featurizer/lm_featurizer.py

```diff
@@ -42,57 +38,83 @@
-        "model_name": "bert",
-        # Pre-Trained weights to be loaded(string)
-        "model_weights": None,
-        # an optional path to a specific directory to download
-        # and cache the pre-trained model weights.
-        "cache_dir": None,
-    }
+    def __init__(
+        self, config: Dict[Text, Any], execution_context: ExecutionContext,
+    ) -> None:
+        """Initializes the featurizer with the model in the config."""
+        super(LanguageModelFeaturizerGraphComponent, self).__init__(
+            execution_context.node_name, config
+        )
+        self._check_language_is_supported()
+        self._load_model_metadata()
+        self._load_model_instance(self._config["skip_model_load"])
+
+    @staticmethod
+    def get_default_config() -> Dict[Text, Any]:
+        """Returns LanguageModelFeaturizer's default config."""
+        return {
+            **DenseFeaturizer2.get_default_config(),
+            # name of the language model to load.
+            "model_name": "bert",
+            # Pre-Trained weights to be loaded(string)
+            "model_weights": None,
+            # an optional path to a specific directory to download
+            # and cache the pre-trained model weights.
+            "cache_dir": None,
+            "skip_model_load": False,
+            "language": None,
+        }
 
     @classmethod
-    def required_components(cls) -> List[Type[Component]]:
-        """Packages needed to be installed."""
-        return [Tokenizer]
+    def validate_config(cls, config: Dict[Text, Any]) -> None:
+        """Validates the configuration."""
+        pass
 
-    def __init__(
-        self,
-        component_config: Optional[Dict[Text, Any]] = None,
-        skip_model_load: bool = False,
+    @classmethod
+    def validate_compatibility_with_tokenizer(
+        cls, config: Dict[Text, Any], tokenizer_type: Type[Tokenizer]
     ) -> None:
-        """Initializes LanguageModelFeaturizer with the specified model.
-
-        Args:
-            component_config: Configuration for the component.
-            skip_model_load: Skip loading the model for pytests.
-        """
-        super(LanguageModelFeaturizer, self).__init__(component_config)
-        self._load_model_metadata()
-        self._load_model_instance(skip_model_load)
+        """Checks that the featurizer and tokenizer are compatible."""
+        pass
 
     @classmethod
     def create(
-        cls, component_config: Dict[Text, Any], config: RasaNLUModelConfig
-    ) -> "DenseFeaturizer":
-        language = config.language
-        if not cls.can_handle_language(language):
-            # check failed
-            raise UnsupportedLanguageError(cls.name, language)
-        return cls(component_config)
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> "LanguageModelFeaturizerGraphComponent":
```

Comment:
Can we please remove all forward references? This requires a `from __future__ import annotations` import at the top to avoid errors
```suggestion
    ) -> LanguageModelFeaturizerGraphComponent:
```

Reply:
Cool. Should we do this for the other components too? It looks like [Mitie](https://github.com/RasaHQ/rasa/blob/a6b0ac95fc70745eef228c0710003d7f6506c2af/rasa/nlu/featurizers/dense_featurizer/mitie_featurizer.py#L66) is using a forward reference, and [Spacy](https://github.com/RasaHQ/rasa/blob/a6b0ac95fc70745eef228c0710003d7f6506c2af/rasa/nlu/featurizers/dense_featurizer/spacy_featurizer.py#L58) is returning a `GraphComponent`.

Reply:
returning a `GraphComponent` (that's the parent class is fine). Great catch with Mitie 🙈